product-mini/platforms/posix/main.c

```diff
@@ -643,14 +643,29 @@ main(int argc, char *argv[])
     }
 #endif
 
-    if (is_repl_mode)
+    ret = 0;
+    if (is_repl_mode) {
         app_instance_repl(wasm_module_inst);
-    else if (func_name)
-        app_instance_func(wasm_module_inst, func_name);
-    else
-        app_instance_main(wasm_module_inst);
+    }
+    else if (func_name) {
+        if (app_instance_func(wasm_module_inst, func_name)) {
+            /* got an exception */
+            ret = 1;
+        }
+    }
+    else {
+        if (app_instance_main(wasm_module_inst)) {
+            /* got an exception */
+            ret = 1;
+        }
+    }
 
-    ret = 0;
+#if WASM_ENABLE_LIBC_WASI != 0
```

Comment:
Just wanted to doublecheck - won't that break any user scenario right now? I guess many scripts assumed the return code is ignored by iwasm, so with the update it's possible this breaks some of the usecases. Just want to highlight we have to definitely include it in the release notes and probably highlight as backward incompatible change - any thoughts @wenyongh @yamt ?

Reply:
Seems that it won't break any user scenarios for us - normally we only use iwasm to run spec cases and standalone cases. Agree to highlight the change in the release notes. And not sure whether it is very important to the users? iwasm just gives a sample, the host embedders can embed WAMR into their application and decide how to handle the exit code.